src/main.cc

```diff
@@ -180,6 +181,35 @@ signal_cb(ev_loop *loop, struct ev_signal *w, int revents)
 	tarantool_exit(0);
 }
 
+static sigint_cb_t internal_sigint_cb = NULL;
+
+static void
+signal_sigint_cb(ev_loop *loop, struct ev_signal *w, int revents)
+{
+	(void) loop;
+	(void) w;
+	(void) revents;
+
+	if (internal_sigint_cb != NULL) {
+		internal_sigint_cb(loop, w, revents);
+		return;
+	}
```

Comment:
Please drop this callback by initializing `sigint_cb` with `signal_cb`:

```C
/*
 * Handle SIGINT like SIGTERM by default, but allow to overwrite the behavior.
 * Used by console.
 */
static sigint_cb_t sigint_cb = signal_cb;

sigint_cb_t
set_sigint_cb(sigint_cb_t cb)
{
	sigint_cb_t old_cb = sigint_cb;
	sigint_cb = cb;
	return old_cb;
}
```

Reply:
This way is not fully implementable. AFAIU the call `ev_signal_init` registers the callback's adress so we can't change it by dynamical replacing the value of `sigint_cb`. I would propose the next:

```C
sigint_cb_t
set_sigint_cb(sigint_cb_t new_sigint_cb)
{
	sigint_cb_t old_cb = signal_sigint_cb;
	ev_signal_init(&ev_sigs[1], new_sigint_cb, SIGINT);
	return old_cb;
}
```

Reply:
I see. Then you should use `ev_set_cb` instead of `ev_signal_init`, because `ev_signal_init` must not be used after `ev_signal_start`.

Reply:
Fixed.